src/main/java/mate/academy/bookstore/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,50 @@
+package mate.academy.bookstore.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.cartitem.CreateCartItemRequestDto;
+import mate.academy.bookstore.dto.cartitem.UpdateCartItemRequestDto;
+import mate.academy.bookstore.dto.shoppingcart.ShoppingCartDto;
+import mate.academy.bookstore.service.ShoppingCartService;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cart")
+@RequiredArgsConstructor
+@PreAuthorize("hasRole('USER')")
```

Comment:
Did you test your code? 

Annotation for specifying a method access-control expression which will be evaluated to decide whether a method invocation is allowed or not.

This annotation works only under a method